compiler/rustc_borrowck/src/diagnostics/conflict_errors.rs

```diff
@@ -445,13 +442,36 @@ impl<'infcx, 'tcx> MirBorrowckCtxt<'_, '_, 'infcx, 'tcx> {
                 } else {
                     (None, &[][..], 0)
                 };
+                let mut reborrow = false;
                 if let Some(def_id) = def_id
                     && let node = self.infcx.tcx.hir_node_by_def_id(def_id)
                     && let Some(fn_sig) = node.fn_sig()
                     && let Some(ident) = node.ident()
                     && let Some(pos) = args.iter().position(|arg| arg.hir_id == expr.hir_id)
                     && let Some(arg) = fn_sig.decl.inputs.get(pos + offset)
```

Comment:
this only works for functions from the current crate, greatly lowering the usefulness of this new diagnostics.

You should instead be able to use `tcx.fn_sig(def_id).skip_binder().skip_binder().inputs()[pos + offset].is_param()`

Reply:
> this only works for functions from the current crate, greatly lowering the usefulness of this new diagnostics.
> 
> You should instead be able to use `tcx.fn_sig(def_id).skip_binder().skip_binder().inputs()[pos + offset].is_param()`

Done. Thank you.
Because `is_param` in https://github.com/rust-lang/rust/blob/c92a8e4d4d7e8da53aa05309ef8d4fac2c9af52f/compiler/rustc_middle/src/ty/sty.rs#L1048 needs a arg `index` which I'm not sure what means, I don't use this function.